src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -338,23 +308,65 @@ protected Task ResolveAuthorityAsync()
             return new KeyValuePair<string, string>?(new KeyValuePair<string, string>(Constants.CCSRoutingHintHeader, OidCCSHeader));
         }
 
+        private void LogRequestStarted(AuthenticationRequestParameters authenticationRequestParameters)
+        {
+            if (authenticationRequestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Info))
+            {
+                int appHashCode = authenticationRequestParameters.AppConfig.GetHashCode();
+                string scopes = authenticationRequestParameters.Scope.AsSingleString();
+                string messageWithPii = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "=== Token Acquisition ({3}) started:\n\tAuthority: {0}\n\tScope: {1}\n\tClientId: {2}\n\tAppHashCode: {4}",
+                    authenticationRequestParameters.AuthorityInfo?.CanonicalAuthority,
+                    scopes,
+                    authenticationRequestParameters.AppConfig.ClientId,
+                    GetType().Name,
+                    appHashCode);
+
+                string messageWithoutPii = string.Format(
+                    CultureInfo.InvariantCulture,
+                    "=== Token Acquisition ({0}) started:\n\t Scopes: {1} AppHashCode: {2}",
+                    GetType().Name,
+                    scopes,
+                    appHashCode);
+
+                if (authenticationRequestParameters.AuthorityInfo != null &&
+                    KnownMetadataProvider.IsKnownEnvironment(authenticationRequestParameters.AuthorityInfo?.Host))
+                {
+                    messageWithoutPii += string.Format(
+                        CultureInfo.CurrentCulture,
+                        "\n\tAuthority Host: {0}",
+                        authenticationRequestParameters.AuthorityInfo?.Host);
+                }
+
+                authenticationRequestParameters.RequestContext.Logger.InfoPii(messageWithPii, messageWithoutPii);
+            }
+
+            if (authenticationRequestParameters.IsConfidentialClient &&
+                !CacheManager.TokenCacheInternal.IsTokenCacheSerialized())
+            {
+                authenticationRequestParameters.RequestContext.Logger.Error("The default token cache provided by MSAL is not designed to be performant when used in confidential client applications. Please use token cache serialization. See https://aka.ms/msal-net-cca-token-cache-serialization.");
+            }
+        }
+
         private void LogReturnedToken(AuthenticationResult result)
         {
             if (result.AccessToken != null && 
                 AuthenticationRequestParameters.RequestContext.Logger.IsLoggingEnabled(LogLevel.Info))
             {
+                int appHashCode = AuthenticationRequestParameters.AppConfig.GetHashCode();
```

Comment:
- Maybe rename here and in log below to appConfigHashCode?
- In which scenario would this hashCode be useful?

Reply:
I think the hash code is useful to know the instance of confidential client being used to track how many CCA are created.

Reply:
Yes, just as Neha stated, it's used to differentiate between different CCA objects. It's to be able to tell if ppl are using 1 or more instances of CCA object, as it has cache implications.